### PR TITLE
Review isprivate

### PR DESCRIPTION
**Pull request by adam-collins**
branched from #244 feature/grails5

### CODE REVIEW

File: grails-app/controllers/au/org/ala/specieslist/EditorController.groovy [hunk 1]
```diff
@@ -95,13 +94,21 @@ class EditorController {
 
         if (params.action == "addRecordScreen") {
             def speciesList = SpeciesList.findByDataResourceUid(params.id)
+            if (!isCurrentUserEditorForList(sl)) {
```
Comment (sughics) [inline grails-app/controllers/au/org/ala/specieslist/EditorController.groovy]: Invalid argument `sli` passed in to isCurrentUserEditorForList. Should be `speciesList`

Comment (sughics) [reply grails-app/controllers/au/org/ala/specieslist/EditorController.groovy]: Updated.

**Review by sughics — APPROVED** [review]

File: grails-app/controllers/au/org/ala/specieslist/SpeciesListController.groovy [hunk 2]
```diff
@@ -307,14 +328,13 @@ class SpeciesListController {
         }
     }
 
-    def test(){
-        log.debug(helperService.addDataResourceForList("My test List")?.toString())
-        log.debug(authService.getEmail()?.toString())
-        //bieService.bulkLookupSpecies(["urn:lsid:biodiversity.org.au:afd.taxon:31a9b8b8-4e8f-4343-a15f-2ed24e0bf1ae"])
-        log.debug(loggerService.getReasons()?.toString())
-    }
-
```
Comment (sughics) [inline grails-app/controllers/au/org/ala/specieslist/SpeciesListController.groovy:316]: This method has been removed but is still referenced in the file. This is throwing errors.

**Comment by sughics** [comment]
Tested a deployed instance of this branch and encountered  a few bugs - included in comments.

File: grails-app/controllers/au/org/ala/specieslist/SpeciesListController.groovy [hunk 3]
```diff
@@ -347,6 +367,12 @@ class SpeciesListController {
             def isdr = params.id.startsWith("dr")
             def guids = getGuidsForList(params.id,grailsApplication.config.downloadLimit)
             def speciesList = isdr?SpeciesList.findByDataResourceUid(params.id):SpeciesList.get(params.id)
+
```
Comment (sughics) [inline grails-app/controllers/au/org/ala/specieslist/SpeciesListController.groovy:371]: `/speciesList/fieldGuide` is invoked when  attempting to download a list and should be invoke-able by a non-authenticated user. This check prevents a non-authenticated user for invoking this method.

**Review by sughics — CHANGES_REQUESTED** [review]

File: grails-app/controllers/au/org/ala/specieslist/SpeciesListController.groovy (hunk 2 above)
Comment (adam-collins) [reply grails-app/controllers/au/org/ala/specieslist/SpeciesListController.groovy:316]: missing method was `getUnmatchedNamesForList`

File: grails-app/controllers/au/org/ala/specieslist/SpeciesListController.groovy [hunk 4]
```diff
@@ -405,9 +427,14 @@ class SpeciesListController {
         if(biocacheService.isListIndexed(params.id)){
             redirect(url:biocacheService.getQueryUrlForList(params.id))
         } else if (params.id && params.type){
+            def splist = SpeciesList.findByDataResourceUid(params.id)
```
[the remark this answers is not in the document]
Comment (adam-collins) [reply grails-app/controllers/au/org/ala/specieslist/SpeciesListController.groovy:440]: changed to `isViewable` method

**Comment by sughics** [comment]
Changes look good - deployed and tested in lists-dev. 
Integration tests now seem to be failing though

**Comment by adam-collins** [comment]
Removed tests that depend on an external auth service

**Review by sughics — APPROVED** [review]

**Review by sughics — APPROVED** [review]

**Pull request merged**